src/test/java/com/hazelcast/wm/test/WebFilterBasicTest.java

```diff
@@ -168,6 +173,40 @@ public void testNoSession() throws Exception {
         assertEquals("true", executeRequest("noSession", serverPort1, cookieStore));
     }
 
+    // https://github.com/hazelcast/hazelcast-wm/issues/15
+    @Test(timeout = 20000)
+ 	public void testInputStreamOfRequestNotConsumed() throws Exception {
+        CookieStore cookieStore = new BasicCookieStore();
+
+        HashMap<String, String> requestParameters = new HashMap<String, String>();
+        requestParameters.put("email", "test@email.com");
+        requestParameters.put("password", "password1");
+
+        assertEquals("true", executeRequest(POST_REQUEST, "login", serverPort1, cookieStore, requestParameters));
+    }
+
+    @Test(timeout = 20000)
+    public void testUseRequestParameterEnabled() throws Exception {
+        String hazelcastSessionId1 = executeRequest(POST_REQUEST, "useRequestParameter", serverPort1,
+                new BasicCookieStore());
+
+        String hazelcastSessionId2 = executeRequest(POST_REQUEST, "useRequestParameter", serverPort2,
+                new BasicCookieStore(), Collections.singletonMap("hazelcast.sessionId", hazelcastSessionId1));
+
+        assertEquals(hazelcastSessionId1, hazelcastSessionId2);
+    }
+
+    @Test(timeout = 20000)
+    public void testUseRequestParameterDisabled() throws Exception {
```

Comment:
I have made changes to `node2-node.xml`which enables `use-request-parameter` for `serverPort2`, leaving `serverPort1` disabled. Making the `POST` request to `serverPort1` which has `use-request-parameter` disabled, this test verifies that the disabled configuration causes generation of a new hazelcast session id.